client/src/main/java/com/tencent/rss/client/impl/ShuffleReadClientImpl.java

```diff
@@ -122,6 +131,12 @@ public CompressedShuffleBlock readShuffleBlockData() {
         return null;
       }
     }
+
+    // All blocks are processed, so just return
+    if (pendingBlockIds.isEmpty()) {
```

Comment:
The check should be called before `if (bufferSegmentQueue.isEmpty()) {`

Reply:
Done